internal/operands/common-templates/reconcile.go

```diff
@@ -113,6 +120,30 @@ func incrementTemplatesRestoredMetric(reconcileResults []common.ReconcileResult,
 	}
 }
 
+func (c *commonTemplates) manageSingleReplicaInfrastructure(request *common.Request) {
+	if request.TopologyMode != v1.SingleReplicaTopologyMode {
+		return
+	}
+
+	for templateIdx, template := range c.templatesBundle {
+		for objIdx, rawObj := range template.Objects {
+			vm := &v12.VirtualMachine{}
+			err := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(rawObj.Raw), 1024).Decode(vm)
+			if err != nil {
+				continue
+			}
+
+			if vm.Spec.Template.Spec.EvictionStrategy != nil && *vm.Spec.Template.Spec.EvictionStrategy == v12.EvictionStrategyLiveMigrate {
+				request.Logger.Info(fmt.Sprintf("Deleting EvictionStrategy from Spec"))
+				vm.Spec.Template.Spec.EvictionStrategy = nil
+				codec := serializer.NewCodecFactory(request.Client.Scheme()).LegacyCodec(v12.GroupVersion)
```

Comment:
Do you have to create NewCodecFactory for each VM?

Reply:
Thanks, my bad. We definitely don't want it.